### PR TITLE
Feature/251 create slack adapter

### PR DESCRIPTION
**Pull request by todd-dembrey**
Create the Slack adapter and re-purpose the messages adapter to present messages to the user when not sending them.

### CODE REVIEW

File: opentech/apply/activity/messaging.py [hunk 1]
```diff
@@ -82,6 +89,56 @@ def send_message(self, message, user, submission, **kwargs):
...
+            slack_target = ''
+
+        message = ' '.join([slack_target, message]).strip()
+        return message
+
+    def slack_id(self, user):
+        return f'<{user.slack}>'
+
+    def send_message(self, message, **kwargs):
+        if not self.destination and not self.target_room:
```
Comment (zerolab) [inline opentech/apply/activity/messaging.py]: Should this not return when either destination or target_room is not set?
i.e. `if not self.destination or not self.target_room:`

File: opentech/apply/determinations/forms.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 from django import forms
 from django.core.exceptions import NON_FIELD_ERRORS
 
-from opentech.apply.funds.workflow import DETERMINATION_RESPONSE_TRANSITIONS
+from opentech.apply.funds.workflow import DETERMINATION_OUTCOMES
```
Comment (zerolab) [inline opentech/apply/determinations/forms.py]: 👍

File: opentech/apply/funds/models.py [hunk 3]
```diff
@@ -46,13 +46,14 @@
 from .edit_handlers import FilteredFieldPanel, ReadOnlyPanel, ReadOnlyInlinePanel
 from .workflow import (
     active_statuses,
+    DETERMINATION_OUTCOMES,
```
Comment (zerolab) [inline opentech/apply/funds/models.py]: imported but not used?

**Comment by todd-dembrey** [comment]
Need a better way to turn off slack notifications during tests. Will add another commit

**Comment by todd-dembrey** [comment]
It was due to my local.py settings rather than the CI tests which was my concern. I'm not too worried if that happens for now.

**Pull request merged**